Jstar1/add upload page

justinwmoore opened the pull request.

Jstar1 committed:
Adding upload page

Jstar1 committed:
adds ability for user to upload images

danieleynis commented:
Could you please add the render call in index.js as part of this PR?

ghost commented:
I can do that

On Thu, Aug 10, 2017 at 6:30 PM Daniel Eynis <notifications@github.com>
wrote:

> Could you please add the render call in index.js as part of this PR?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/danieleynis/Hysh/pull/28#issuecomment-321714389>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/APHlbyY5NCxJz32Fabgu3dfhm7gVrjhXks5sW67KgaJpZM4Oy3n->
> .
>

danieleynis commented:
Please do it on this branch.

ghost commented:
Word

On Thu, Aug 10, 2017 at 6:43 PM Daniel Eynis <notifications@github.com>
wrote:

> Please do it on this branch.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/danieleynis/Hysh/pull/28#issuecomment-321715585>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/APHlb2-VrthWjczsZIzADnfXO4gzd0vOks5sW7G5gaJpZM4Oy3n->
> .
>

danieleynis commented:
Since Justin is the dev owner for this, just talk to him first about working on his branch/his feature so there is no confusion.

justinwmoore commented:
I'm fine with this

Commit:
added rendering for upload.njk

Commit:
Update index.js

commenting correction

ghost commented:
Ok i added rendering

danieleynis commented:
@RuddJohnson you added the route for `/upload` but did not remove the `/upload` route that was already there.  You could have just edited that one. Please either remove the other one or remove the one you added and edit the existing one. I put an explicit TODO there from a previous PR.

Lines are located at: https://github.com/danieleynis/Hysh/blob/jstar1/add-upload-page/routes/index.js#L80-L82

danieleynis commented on views/upload.njk:
```diff
@@ -0,0 +1,24 @@
+</div>
+{% endblock %}
+{% block script%}
+<script src="/js/upload.js"></script>
```
`/js/upload.js` does not seem to exist, I am getting errors because it cannot retrieve that file. Do you plan on adding it later? If it is required please add it as part of this branch and PR.

danieleynis commented on views/upload.njk:
```diff
@@ -0,0 +1,24 @@
+  <div class="span12">
+    <h1>Upload file here!</h1>         
+    <div class="centerupload">
+     <form action="/upload" id="form1">
```
This might `form` tag might require a `method="post"` attribute

danieleynis commented on routes/index.js:
```diff
@@ -57,6 +57,11 @@ module.exports = function(passport){
   });
 
+  /*Nunjucks rendering for upload page*/
+  router.get('/upload',function(req,res){
```
`router.get('/upload'...` already exists in the file. Please either delete the other one or this one and modify the existing one.

Commit:
addressing reviews

Commit:
Merge branch 'jstar1/add-upload-page' of https://github.com/danieleynis/Hysh into jstar1/add-upload-page

ghost commented:
addressed feedback

Jstar1 committed:
Adds posting method for form

Jstar1 committed:
Adds jquery to handle files

Added the jquery so files can be loaded from user disk
Made js directory for storing javascript files

Jstar1 committed:
Adds rendering for upload page

Jstar1 committed:
Adds css styling for upload page

justinwmoore commented:
Hey I want to apologize for the incomplete code i sent in for a pr without thoroughly going through my code and validating it myself, (A bit embarrassing and unprofessional, I should know better.). I am sorry for the confusion, thank you Danny for blocking this before it got merged. Sorry Rudd for giving you the 'okay' to work on some code that was incomplete there for confusing. 

Once again my apologizes. 

Justin-

Commit:
Merge branch 'jstar1/add-upload-page' of https://github.com/danieleynis/Hysh into jstar1/add-upload-page

Commit:
resolving duplicate upload rendering

danieleynis commented:
@jstar1 Doing some testing right now. When I select an image it display the image I selected on the UI but if the image is large it pushed the upload button down and I cant scroll down to click upload. See screenshot: 


![image](https://user-images.githubusercontent.com/11580217/29244073-209a3ba6-7f64-11e7-8b63-4d6bdda1362e.png)

justinwmoore commented:
I had a hard time trying to resize images, if you have any input let me know

justinwmoore commented:
can you test smaller images?

danieleynis commented:
Is there anyway to resize the image using jQuery?

https://stackoverflow.com/questions/1143517/jquery-resizing-image

danieleynis commented:
The real issue here I think is that I can't even scroll down to click upload.

justinwmoore commented:
Yes this can be done, Rudd is going to give it a shot.

danieleynis commented:
@RuddJohnson Just let me know when this is resolved.

danieleynis commented on views/upload.njk:
```diff
@@ -0,0 +1,31 @@
+    <h1>Upload file here!</h1>         
+    <div class="centerupload">
+     <form action="/upload" id="form1" method="POST">
+      <input type='file' id="imgInp" accept="file"/*>
```
`accept="file"/*` I think this is a syntax error (`/*` is outside quotes). I think this must be `accept="image/*"`. You can find out more here: https://www.w3schools.com/tags/tryit.asp?filename=tryhtml_input_accept

danieleynis replied:
In addition to this you must specify a `name` attribute,  specifically `name="picture"`

Commit:
added image scaling to upload page

danieleynis commented on views/upload.njk:
```diff
@@ -16,9 +23,11 @@
     <div class="centerupload">
      <form action="/upload" id="form1" method="POST">
-      <input type='file' id="imgInp" accept="file"/*>
+      <input type='file' id="imgInp" accept="image"/*>
```
There is still a syntax error the `/*` is outside the parens!

danieleynis replied:
Also there is still no `name` attribute. Please look and review all my comments carefully.

danieleynis commented:
Back to you.

ghost commented:
What parens? There are no parenthesis.


On Sat, Aug 12, 2017 at 3:44 PM Daniel  <notifications@github.com> wrote:

> Back to you.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/danieleynis/Hysh/pull/28#issuecomment-322010395>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/APHlb82lfS27xGzb4BOK7OcDjVruwkcRks5sXirRgaJpZM4Oy3n->
> .
>

ghost commented:
You mean quotations?

On Sat, Aug 12, 2017 at 3:46 PM Rudd Johnson <rujo2@pdx.edu> wrote:

> What parens? There are no parenthesis.
>
>
> On Sat, Aug 12, 2017 at 3:44 PM Daniel  <notifications@github.com> wrote:
>
>> Back to you.
>>
>> —
>> You are receiving this because you were mentioned.
>>
>>
>> Reply to this email directly, view it on GitHub
>> <https://github.com/danieleynis/Hysh/pull/28#issuecomment-322010395>, or mute
>> the thread
>> <https://github.com/notifications/unsubscribe-auth/APHlb82lfS27xGzb4BOK7OcDjVruwkcRks5sXirRgaJpZM4Oy3n->
>> .
>>
>

Commit:
added name attribute

danieleynis commented:
Yes sorry meant quotes. Derp.

danieleynis committed:
Updated upload.njk

* Added type="submit" to upload button (required)
* Removed method="POST" (my misunderstanding this is actually not required and was causing issues for some reason)

danieleynis commented:
I updated the necessary file to finally make it work.
